src/main/java/org/broadinstitute/hellbender/utils/pileup/PileupElement.java

```diff
@@ -317,45 +317,69 @@ public int getOffsetInCurrentCigar() {
         return offsetInCurrentCigar;
     }
 
-    /** for some helper functions */
-    private enum Direction { PREV, NEXT }
+    /**  Direction of search for some helper functions */
+    @VisibleForTesting
+    enum Direction { PREV(-1), NEXT(1);
+        private final int increment;
+
+        public int getIncrement() {
+            return increment;
+        }
+
+        private Direction(final int increment){
+            this.increment = increment;
+        }
+    }
 
     /**
      * Helper function to get cigar elements between this and either the prev or next genomic position
      *
      * @param direction PREVIOUS if we want before, NEXT if we want after
      * @return a non-null list of cigar elements between this and the neighboring position in direction
      */
-    private LinkedList<CigarElement> getBetween(final Direction direction) {
-        final int increment = direction == Direction.NEXT ? 1 : -1;
-        LinkedList<CigarElement> elements = null;
+    private List<CigarElement> getBetween(final Direction direction) {
```

Comment:
Document the behavior of this confusing method -- especially the fact that it stops accumulating elements at the first "on-genome" operator it encounters